screens/Authentication/SignUp/CreateCredentials.tsx

```diff
@@ -0,0 +1,189 @@
+import CredentialsButton from "@/components/CredentialsButton"
+import Badge from "@/components/ui/Badge"
+import CustomText from "@/components/ui/CustomText"
+import Input from "@/components/ui/Input"
+import CrossMark from "@/components/vectors/CrossMark"
+import { VALIDATION_PATTERNS } from "@/constants/patterns"
+import { RootStackParamList } from "@/navigation"
+import { theme } from "@/theme"
+import { AccountCreationStep } from "@/types"
+import { useNavigation } from "@react-navigation/native"
+import { NativeStackNavigationProp } from "@react-navigation/native-stack"
+import { useEffect, useMemo, useState } from "react"
+import {
+  StyleSheet,
+  View,
+  TouchableOpacity,
+  KeyboardAvoidingView,
+  Platform,
+} from "react-native"
+
+type CredentialsScrenProps = NativeStackNavigationProp<
+  RootStackParamList,
+  "EditSignal"
+>
+
+export default function CreateCredentials() {
+  const navigation = useNavigation<CredentialsScrenProps>()
+  const [step, setStep] = useState<AccountCreationStep>(1)
```

Comment:
if we're gonna add return to the previous step, we might need to handle the text better than this. let's discuss the requirements